### PR TITLE
Job Status Enhancements

### PR DESCRIPTION
**Pull request by cweibel**
- An API endpoint `v1/status/jobs` was added to show the most recent success or failure of a job task.  Intended for a monitoring/alerting application to scrape this information periodically
 - In GUI, under Jobs the status is now color coded
 - Also on the dashboard recently failed jobs will appear in a notification bar at the top

### CODE REVIEW

File: docs/api/http.md [hunk 1]
```diff
@@ -63,6 +64,32 @@ The top-level keys are:
 not rely on the contents, semantics, or format of the response
 body.  The documentation presented here is informative only.
 
+#### GET /v1/status/jobs
+
+This API emits information about the success or failure of the
+most recent task associated with a job.  This information is
+intended to be used with a monitoring system to alert on job health.
+
+```json
```
Comment (lnguyen) [inline docs/api/http.md]: is this returning array or just a single job?

Comment (cweibel) [inline docs/api/http.md]: When there is more than one job this is an example of the output:
```
{
  "Another job": {
    "name": "Another job",
    "last_run": 0,
    "next_run": 1492705500,
    "paused": false,
    "status": "pending"
  },
  "My job": {
    "name": "My job",
    "last_run": 1492701900,
    "next_run": 1492705500,
    "paused": false,
    "status": "done"
  }
}
```

Comment (lnguyen) [inline docs/api/http.md]: ok just curious

File: db/jobs.go [hunk 2]
```diff
@@ -48,11 +53,11 @@ type JobFilter struct {
...
 
 func (f *JobFilter) Query() (string, []interface{}) {
-	wheres := []string{"j.uuid = j.uuid"}
+	var wheres []string = []string{"j.uuid = j.uuid"}
 	var args []interface{}
 	if f.SearchName != "" {
-		comparator := "LIKE"
-		toAdd := Pattern(f.SearchName)
+		var comparator string = "LIKE"
+		var toAdd string = Pattern(f.SearchName)
```
Comment (jhunt) [inline db/jobs.go]: These changes need to be reverted

File: db/jobs.go [hunk 3]
```diff
@@ -118,14 +131,92 @@ func (db *DB) GetAllJobs(filter *JobFilter) ([]*Job, error) {
...
+			ann.LastTaskStatus = "pending"
+		}
+
+		l = append(l, ann)
+	}
+
+	return l, nil
+}
+
+func (db *DB) GetAllJobsStatus(filter *JobFilter) ([]*Job, error) {
```
Comment (jhunt) [inline db/jobs.go]: GetAllJobsStatus was implemented but never used, because we found better SQL

File: db/jobs.go [hunk 4]
```diff
@@ -86,17 +91,25 @@ func (f *JobFilter) Query() (string, []interface{}) {
 	}
 
 	return `
+	WITH most_recent_job_task_at AS (
+		SELECT job_uuid,  max(requested_at) AS requested_at FROM tasks group by job_uuid
+	),
+	most_recent_job_task AS (
+		SELECT t.started_at, t.status, t.job_uuid, t.uuid FROM tasks t, most_recent_job_task_at mr WHERE t.job_uuid = mr.job_uuid AND t.requested_at = mr.requested_at
+	)
```
Comment (jhunt) [inline db/jobs.go]: This won't work for MySQL, so maybe we should just omit the status check and MySQL implementations won't get monitoring?

**Pull request merged**